api/envoy/extensions/upstreams/http/v3/http_protocol_options.proto

```diff
@@ -75,6 +76,15 @@ message HttpProtocolOptions {
     config.core.v3.Http2ProtocolOptions http2_protocol_options = 2;
   }
 
+  // If this is used, the cluster can will use both HTTP/1 and HTTP/2, whichever
+  // protocol is negotiated by ALPN with the upstream.
+  // If the upstream does not support ALPN, it will fail over to HTTP/1.
```

Comment:
If we expect to negotiate using ALPN and the transport doesn't support ALPN, it may be confusing to the user to fall back to HTTP/1 without explicit configuration. Do you think this should be a config error by default?

Reply:
The transport must support ALPN.  The endpoint may not - when talking to legacy servers, if they don't do ALPN we need to use HTTP/1 for this to work on the internet.

I'll add more comments to try to make the distinction clear!

Reply:
Yes that makes sense, thanks.